pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -118,11 +120,35 @@ func (pl *VolumeBinding) EventsToRegister() []framework.ClusterEventWithHint {
 		// When CSIStorageCapacity is enabled, pods may become schedulable
 		// on CSI driver & storage capacity changes.
 		{Event: framework.ClusterEvent{Resource: framework.CSIDriver, ActionType: framework.Add | framework.Update}},
-		{Event: framework.ClusterEvent{Resource: framework.CSIStorageCapacity, ActionType: framework.Add | framework.Update}},
+		{Event: framework.ClusterEvent{Resource: framework.CSIStorageCapacity, ActionType: framework.Add | framework.Update}, QueueingHintFn: pl.isSchedulableAfterCSIStorageCapacityChange},
 	}
 	return events
 }
 
+func (pl *VolumeBinding) isSchedulableAfterCSIStorageCapacityChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	_, newCap, err := util.As[*storagev1.CSIStorageCapacity](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	logger = klog.LoggerWithValues(
+		logger,
+		"Pod", klog.KObj(pod),
+		"CSIStorageCapacity", klog.KObj(newCap),
+	)
+
+	for _, vol := range pod.Spec.Volumes {
+		if vol.PersistentVolumeClaim != nil || vol.Ephemeral != nil {
+			// This Pod might have got unschedulable due to CSIStorageCapacity in a past scheduling cycle.
+			logger.V(5).Info("CSIStorageCapacity was created or updated, but the pod doesn't mount any persistent volumes and generic ephemeral storage volumes. It doesn't make this pod schedulable.")
```

Comment:
It makes me wonder: in the first place, could a Pod without any PVC or ephemeral volume get unschedulable by VolumeBinding plugin?
If No, maybe we end up having no QHint for `CSIStorageCapacity` (for now, at least until we build some observability [like I said](https://github.com/kubernetes/kubernetes/pull/124958#discussion_r1615093543)).

Reply:
I'm sorry, I made a mistake in correcting the log message. 
(This has been fixed)

> It makes me wonder: in the first place, could a Pod without any PVC or ephemeral volume get unschedulable by VolumeBinding plugin?

I think the answer to this is No.

> If No, we likely end up having no QHint for CSIStorageCapacity (for now, at least until we build some observability like I said).

Regarding this point, I think that if we don't return QueueSkip in the QHint, scheduling will be attempted due to CSIStorageCapacity changes even in cases where the Pod doesn't have a Volume. Would this not cause any issues in your opinion?

Reply:
> Regarding this point, I think that if we don't return QueueSkip in the QHint, scheduling will be attempted due to CSIStorageCapacity changes even in cases where the Pod doesn't have a Volume. Would this not cause any issues in your opinion?

QHint is executed for Pods which was failed by plugins. e.g., if, in a previous scheduling cycle, a Pod is failed by resource fit plugin and PodAffinity plugin, then only QHints of those plugins would be run for the Pod.

Then, if the answer to my first question "could a Pod without any PVC or ephemeral volume get unschedulable by VolumeBinding plugin?" is No, filtering Pods with volume in QHint doesn't make any point since only Pods with volume would reach this QHint in the first place.

Reply:
Thank you for your input. I understand.
I'll proceed to close the unnecessary PRs.